subsys/llext/llext.c

```diff
@@ -266,18 +304,37 @@ static int llext_copy_section(struct llext_loader *ldr, struct llext *ext,
 	    IS_ENABLED(CONFIG_LLEXT_STORAGE_WRITABLE)) {
 		ext->mem[mem_idx] = llext_peek(ldr, ldr->sects[mem_idx].sh_offset);
 		if (ext->mem[mem_idx]) {
+			llext_init_mem_part(ext, mem_idx, (uintptr_t)ext->mem[mem_idx],
+				ldr->sects[mem_idx].sh_size);
 			ext->mem_on_heap[mem_idx] = false;
 			return 0;
 		}
 	}
 
-	ext->mem[mem_idx] = k_heap_aligned_alloc(&llext_heap, sizeof(uintptr_t),
-						 ldr->sects[mem_idx].sh_size,
+#ifndef CONFIG_ARM_MPU
+	const uintptr_t sect_alloc = ROUND_UP(ldr->sects[mem_idx].sh_size, LLEXT_PAGE_SIZE);
+	const uintptr_t sect_align = LLEXT_PAGE_SIZE;
+#else
```

Comment:
Why is ARM MPU handled differently? Maybe some note so that anyone touching this in the future can keep it in mind?

Reply:
Yep ... These alignments constraints are somehow already handled by userspace code, maybe we need to export this logic in future.

Reply:
I'd welcome a conveniently wrapped mmu page/mpu region allocator, I'd note the permission bits need to be set *after* copying of data occurs in this case.